pallets/xst/src/lib.rs

```diff
@@ -181,12 +176,79 @@ pub mod pallet {
 
         #[pallet::weight(<T as Config>::WeightInfo::enable_synthetic_asset())]
         pub fn enable_synthetic_asset(
+            origin: OriginFor<T>,
+            asset_symbol: AssetSymbol,
+            asset_name: AssetName,
+            reference_symbol: T::Symbol,
+            fee_ratio: Fixed,
+        ) -> DispatchResultWithPostInfo {
+            ensure_root(origin)?;
+
+            Self::enable_synthetic_asset_unchecked(
+                asset_symbol,
+                asset_name,
+                reference_symbol,
+                fee_ratio,
+                true,
+            )?;
+            Ok(().into())
+        }
+
+        /// Disable synthetic asset.
+        ///
+        /// Just remove synthetic from exchanging.
+        /// Will not unregister trading pair because `trading_pair` pallet does not provide this
+        /// ability. And will not unregister trading synthetic asset because of that.
+        ///
+        /// - `origin`: the sudo account on whose behalf the transaction is being executed,
+        /// - `synthetic_asset`: synthetic asset id to disable.
+        #[pallet::weight(<T as Config>::WeightInfo::disable_synthetic_asset())]
+        pub fn disable_synthetic_asset(
             origin: OriginFor<T>,
             synthetic_asset: T::AssetId,
         ) -> DispatchResultWithPostInfo {
             ensure_root(origin)?;
-            EnabledSynthetics::<T>::mutate(|set| set.insert(synthetic_asset));
-            Self::deposit_event(Event::SyntheticAssetEnabled(synthetic_asset));
+
+            let reference_symbol = EnabledSynthetics::<T>::try_get(synthetic_asset)
+                .map_err(|_| Error::<T>::SyntheticIsNotEnabled)?
+                .expect("Synthetic tables always store `Some`")
+                .reference_symbol;
```

Comment:
```suggestion
            let reference_symbol = EnabledSynthetics::<T>::get(synthetic_asset)
                .ok_or(|_| Error::<T>::SyntheticIsNotEnabled)?
                .reference_symbol;
```
Should work like this with OptionQuery